lib/models/userModel.ts

```diff
@@ -0,0 +1,49 @@
+import { Schema, model, Document, Model, Mongoose } from 'mongoose';
+enum roles{
+    user = "user",
+    bq = "bierquaestor",
+    hbq = "hauptbierquaestor",
+    admin = "admin"
+}
+interface IUser extends Document{
+    firstname: string;
+    lastname: string;
+    email: string;
+    password: string;
+    role : string; // enum : ["user" , "bq", "hbq", "admin"]
+    ownedCustomer : Schema.Types.ObjectId;
+}
+
+interface UserModel extends Model<IUser>{};
+
+class User {
+    
+    private _model: Model<IUser>;
+
+    constructor() {
+        const schema =  new Schema({
```

Comment:
Wie fändest du es, die Logic in eine Methode auszulagern wie z.B. initModel und diese dann im constructor aufzurufen?

Reply:
Kann man machen, allerdings würde ich den Schritt erst gehen, sobald mehr als eine Aufgabe im Konstruktor erledigt wird. Nicht sicher, welche das sein sollte